lib_eio_linux/eio_stubs.c

```diff
@@ -99,3 +99,10 @@ CAMLprim value caml_eio_getdents(value v_fd) {
 
   CAMLreturn(result);
 }
+
+CAMLextern int caml_convert_signal_number(int);
+
+CAMLprim value caml_signal_to_posix_signal(value v_signo)
```

Comment:
What is this for?

Reply:
I should have made that clearer in the code sorry. The problem is that the OCaml signal numbers don't match the POSIX ones (or at least the ones here: https://man7.org/linux/man-pages/man7/signal.7.html). That's why earlier commits were failing with:

```
Full test results in `/src/_build/default/lib_eio_linux/tests/_build/_tests/eio_linux'.
--
355 | Test Successful in 0.003s. 5 tests run.
356 | File "tests/process.md", line 1, characters 0-0:
357 | diff --git a/_build/default/tests/process.md b/_build/default/tests/.mdx/process.md.corrected
358 | index ade46dd..1eb6ef1 100644
359 | --- a/_build/default/tests/process.md
360 | +++ b/_build/default/tests/.mdx/process.md.corrected
361 | @@ -44,7 +44,7 @@ Stopping a subprocess works and checking the status waits and reports correctly
362 | let t = spawn ~sw "sleep" [ "sleep"; "10" ] in
363 | Process.stop t;
364 | Promise.await (Process.status t);;
365 | -- : Process.status = Eio.Process.Signaled 9
366 | +- : Process.status = Eio.Process.Signaled (-7)
367 | ```
368 |  
369 | A switch will stop a process when it is released.
370 | @@ -59,7 +59,7 @@ A switch will stop a process when it is released.
371 | in
372 | run ();
373 | Promise.await @@ Process.status (Option.get !proc);;
374 | -- : Process.status = Eio.Process.Signaled 9
375 | +- : Process.status = Eio.Process.Signaled (-7)
```

as I ran the tests using Luv (which doesn't use OCaml signal numbers). So I was converting them to be the same. Do you think there's a better approach?

Reply:
I don't think there is a standard set of numbers. Even the Linux man-page you linked uses different numbers for different architectures. Seems tempting to use a polymorphic variant here instead (e.g. ``Eio.Process.Signaled `TERM``.

Reply:
I had started this when I was playing with signals:
https://github.com/haesbaert/eio/blob/signal/lib_eio/include/discover.ml
Basically exports the SIGFOO and also tries to compile a program with the non standard SIGBARS that get exported as an optional. It might make sense to point out also that the Stdlib signal calls work with both the ocaml values (the negative ones) and the "real ones".